<a href="https://colab.research.google.com/github/sasikanth6076/Python-EDA-with-libraries/blob/main/Pandas_Data_Cleaning_and_Manipulation_Techniques_7th_july_6_00pm.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Problem statement
- **Skill/location based job search**

# **Importing Dataset from Github**

In [1]:
!git clone https://github.com/AshishJangra27/datasets

fatal: destination path 'datasets' already exists and is not an empty directory.


In [2]:
import pandas as pd
df = pd.read_csv('/content/datasets/Job Postings/jobs.csv.zip')

## 1. **Data** **Exploration**

In [3]:
df

job_id                                           job_role  \
0      7.012301e+10      Branch Banking - Calling For Women Candidates   
1      6.012391e+10                       Product Owner Senior Manager   
2      6.012391e+10  Employee Relations and Policies Associate Manager   
3      6.012391e+10         Employee Relations and Policies Specialist   
4      6.012301e+10                                  SAP BO Consultant   
...             ...                                                ...   
79142  6.012301e+10                           .Net Fullstack Developer   
79143  1.122201e+10        Senior (BI) Business Intelligence Architect   
79144  1.512220e+11                                  PowerBI Developer   
79145  2.712220e+11                         Customer Service Associate   
79146  2.112220e+11                   Senior Java BED ( PAN India)- KG   

                     company experience                     salary  \
0                  Hdfc Bank    1-6 Yrs              Not disclosed   
1                  Accenture  11-15 Yrs              Not disclosed   
2                  Accenture    3-7 Yrs              Not disclosed   
3                  Accenture    3-7 Yrs              Not disclosed   
4                   Mindtree    5-7 Yrs              Not disclosed   
...                      ...        ...                        ...   
79142             Citiustech    3-8 Yrs   7,00,000 - 17,00,000 PA.   
79143  Hexaware Technologies  10-13 Yrs              Not disclosed   
79144                     EY   5-10 Yrs              Not disclosed   
79145          Tech Mahindra    0-2 Yrs    2,50,000 - 3,50,000 PA.   
79146          Tech Mahindra   8-12 Yrs  25,00,000 - 27,50,000 PA.   

                                                location  rating  \
0      Kolkata, Hyderabad/Secunderabad, Pune, Ahmedab...     4.0   
1      Kolkata, Mumbai, Hyderabad/Secunderabad, Pune,...     4.1   
2      Kolkata, Mumbai, Hyderabad/Secunderabad, Pune,...     4.1   
3      Kolkata, Mumbai, Hyderabad/Secunderabad, Pune,...     4.1   
4      Hybrid - Kolkata, Hyderabad/Secunderabad, Pune...     4.1   
...                                                  ...     ...   
79142  Hybrid - Pune, Bangalore/Bengaluru, Mumbai (Al...     4.0   
79143  Hybrid - Pune, Chennai, Bangalore/Bengaluru, M...     3.7   
79144                  Hybrid - Pune, Mumbai (All Areas)     3.8   
79145  Temp. WFH - Noida, Kolkata, New Delhi, Hyderab...     3.7   
79146  Temp. WFH - Noida, Kolkata, Hyderabad/Secunder...     3.7   

             reviews                                    resposibilities  \
0      39110 Reviews     Customer Service,Sales,Relationship Management   
1      32129 Reviews  Product management,Market analysis,Change mana...   
2      32129 Reviews  Business process,Change management,Team manage...   
3      32129 Reviews  Business process,Change management,Team manage...   
4       3759 Reviews  SAP BO,PL / SQL,Oracle SQL,SAP Business Object...   
...              ...                                                ...   
79142    742 Reviews  c#,.net core,javascript,Microservices,Angular,...   
79143   3172 Reviews  Informatica Mdm,etl,business intelligence,Agil...   
79144   6267 Reviews                            pagination,Power Bi,Dax   
79145  21440 Reviews  Bengali,International BPO,BPO,Customer Service...   
79146  21440 Reviews  Kafka,sql,spring boot,REST API,java,Lambda,Des...   

         posted_on                                           job_link  \
0        1 Day Ago  https://www.naukri.com/job-listings-branch-ban...   
1        1 Day Ago  https://www.naukri.com/job-listings-product-ow...   
2        1 Day Ago  https://www.naukri.com/job-listings-employee-r...   
3        1 Day Ago  https://www.naukri.com/job-listings-employee-r...   
4        1 Day Ago  https://www.naukri.com/job-listings-sap-bo-con...   
...            ...                                                ...   
79142   2 Days Ago  https://www.naukri.com/job-listing

In [4]:
df.head()

job_id                                           job_role    company  \
0  7.012301e+10      Branch Banking - Calling For Women Candidates  Hdfc Bank   
1  6.012391e+10                       Product Owner Senior Manager  Accenture   
2  6.012391e+10  Employee Relations and Policies Associate Manager  Accenture   
3  6.012391e+10         Employee Relations and Policies Specialist  Accenture   
4  6.012301e+10                                  SAP BO Consultant   Mindtree   

  experience         salary  \
0    1-6 Yrs  Not disclosed   
1  11-15 Yrs  Not disclosed   
2    3-7 Yrs  Not disclosed   
3    3-7 Yrs  Not disclosed   
4    5-7 Yrs  Not disclosed   

                                            location  rating        reviews  \
0  Kolkata, Hyderabad/Secunderabad, Pune, Ahmedab...     4.0  39110 Reviews   
1  Kolkata, Mumbai, Hyderabad/Secunderabad, Pune,...     4.1  32129 Reviews   
2  Kolkata, Mumbai, Hyderabad/Secunderabad, Pune,...     4.1  32129 Reviews   
3  Kolkata, Mumbai, Hyderabad/Secunderabad, Pune,...     4.1  32129 Reviews   
4  Hybrid - Kolkata, Hyderabad/Secunderabad, Pune...     4.1   3759 Reviews   

                                     resposibilities  posted_on  \
0     Customer Service,Sales,Relationship Management  1 Day Ago   
1  Product management,Market analysis,Change mana...  1 Day Ago   
2  Business process,Change management,Team manage...  1 Day Ago   
3  Business process,Change management,Team manage...  1 Day Ago   
4  SAP BO,PL / SQL,Oracle SQL,SAP Business Object...  1 Day Ago   

                                            job_link  \
0  https://www.naukri.com/job-listings-branch-ban...   
1  https://www.naukri.com/job-listings-product-ow...   
2  https://www.naukri.com/job-listings-employee-r...   
3  https://www.naukri.com/job-listings-employee-r...   
4  https://www.naukri.com/job-listings-sap-bo-con...   

                                        company_link  
0  https://www.naukri.com/hdfc-bank-jobs-careers-213  
1  https://www.naukri.com/accenture-jobs-careers-...  
2  https://www.naukri.com/accenture-jobs-careers-...  
3  https://www.naukri.com/accenture-jobs-careers-...  
4  https://www.naukri.com/mindtree-jobs-careers-3...

In [5]:
df.tail()

job_id                                     job_role  \
79142  6.012301e+10                     .Net Fullstack Developer   
79143  1.122201e+10  Senior (BI) Business Intelligence Architect   
79144  1.512220e+11                            PowerBI Developer   
79145  2.712220e+11                   Customer Service Associate   
79146  2.112220e+11             Senior Java BED ( PAN India)- KG   

                     company experience                     salary  \
79142             Citiustech    3-8 Yrs   7,00,000 - 17,00,000 PA.   
79143  Hexaware Technologies  10-13 Yrs              Not disclosed   
79144                     EY   5-10 Yrs              Not disclosed   
79145          Tech Mahindra    0-2 Yrs    2,50,000 - 3,50,000 PA.   
79146          Tech Mahindra   8-12 Yrs  25,00,000 - 27,50,000 PA.   

                                                location  rating  \
79142  Hybrid - Pune, Bangalore/Bengaluru, Mumbai (Al...     4.0   
79143  Hybrid - Pune, Chennai, Bangalore/Bengaluru, M...     3.7   
79144                  Hybrid - Pune, Mumbai (All Areas)     3.8   
79145  Temp. WFH - Noida, Kolkata, New Delhi, Hyderab...     3.7   
79146  Temp. WFH - Noida, Kolkata, Hyderabad/Secunder...     3.7   

             reviews                                    resposibilities  \
79142    742 Reviews  c#,.net core,javascript,Microservices,Angular,...   
79143   3172 Reviews  Informatica Mdm,etl,business intelligence,Agil...   
79144   6267 Reviews                            pagination,Power Bi,Dax   
79145  21440 Reviews  Bengali,International BPO,BPO,Customer Service...   
79146  21440 Reviews  Kafka,sql,spring boot,REST API,java,Lambda,Des...   

         posted_on                                           job_link  \
79142   2 Days Ago  https://www.naukri.com/job-listings-net-fullst...   
79143    1 Day Ago  https://www.naukri.com/job-listings-senior-bi-...   
79144   2 Days Ago  https://www.naukri.com/job-listings-powerbi-de...   
79145  10 Days Ago  https://www.naukri.com/job-listings-customer-s...   
79146   3 Days Ago  https://www.naukri.com/job-listings-senior-jav...   

                                            company_link  
79142  https://www.naukri.com/citiustech-jobs-careers...  
79143  https://www.naukri.com/hexaware-technologies-j...  
79144        https://www.naukri.com/ey-jobs-careers-9156  
79145  https://www.naukri.com/tech-mahindra-jobs-care...  
79146  https://www.naukri.com/born-commerce-jobs-care...

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 79147 entries, 0 to 79146
Data columns (total 12 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   job_id           78667 non-null  float64
 1   job_role         78667 non-null  object 
 2   company          78666 non-null  object 
 3   experience       77398 non-null  object 
 4   salary           78667 non-null  object 
 5   location         77441 non-null  object 
 6   rating           42948 non-null  float64
 7   reviews          42948 non-null  object 
 8   resposibilities  78647 non-null  object 
 9   posted_on        78667 non-null  object 
 10  job_link         78667 non-null  object 
 11  company_link     78667 non-null  object 
dtypes: float64(2), object(10)
memory usage: 7.2+ MB


In [7]:
df.describe()

job_id        rating
count  7.866700e+04  42948.000000
mean   1.255071e+11      3.940144
std    1.097464e+11      0.519211
min    1.012200e+10      1.000000
25%    4.012350e+10      3.700000
50%    6.012350e+10      4.000000
75%    2.610226e+11      4.200000
max    3.112229e+11      5.000000

In [8]:
df.columns

Index(['job_id', 'job_role', 'company', 'experience', 'salary', 'location',
       'rating', 'reviews', 'resposibilities', 'posted_on', 'job_link',
       'company_link'],
      dtype='object')

In [9]:
df['job_id']

0        7.012301e+10
1        6.012391e+10
2        6.012391e+10
3        6.012391e+10
4        6.012301e+10
             ...     
79142    6.012301e+10
79143    1.122201e+10
79144    1.512220e+11
79145    2.712220e+11
79146    2.112220e+11
Name: job_id, Length: 79147, dtype: float64

In [10]:
df['job_role'].unique()

array(['Branch Banking - Calling For Women Candidates',
       'Product Owner Senior Manager',
       'Employee Relations and Policies Associate Manager', ...,
       'Hiring For International Voice Process | Gurgaon | 35k',
       'Fresher Engineer', 'Anatomy Trainer - Part Time / Full time'],
      dtype=object)

In [11]:
len(df['job_role'].unique())

51051

In [12]:
df['job_role'].duplicated().sum()

np.int64(28096)

In [13]:
df['job_role'].value_counts().head(25)

job_role
Business Development Executive                                       305
Solution Architect                                                   278
Customer Service Associate                                           277
Sales Executive                                                      275
.Net Fullstack Developer                                             258
People Advisor Senior Analyst                                        256
SAP BO Consultant                                                    255
PowerBI Developer                                                    254
Senior (BI) Business Intelligence Architect                          254
Senior Product Experience Engineer                                   254
Unified Functional Testing/UFT Automation Testing                    254
Business Development Manager                                         254
Hiring-Java-Pune/Mumbai/Bangalore/Hyderabad/Chennai/Gurgaon/Noida    254
Data Science Specialist                                              254
Employee Relations and Policies Specialist                           254
ServiceNow Consultants / Sr. Consultants ( Development )             254
Product Owner Senior Manager                                         254
Functional Architect -  Oracle Supply Chain Management (Fusion)      254
HR Service Delivery Senior Analyst                                   254
Senior Java BED ( PAN India)- KG                                     254
Branch Banking - Calling For Women Candidates                        254
Employee Relations and Policies Associate Manager                    254
Solution Designer                                                    254
Java Developer                                                       252
Sales Manager                                                        158
Name: count, dtype: int64

In [14]:
df['job_role'].value_counts().tail(25)

job_role
CAE - Multibody Dynamics                                                  1
Administrator - IT & Facility Management                                  1
Data Warehouse Consultant                                                 1
Consultant ( IT INFRA )                                                   1
Communication Skills                                                      1
Associate IAM                                                             1
Application Support Role                                                  1
Plant Purchase Co-ordinator                                               1
Senior Associate - Streaming                                              1
APS Engineer                                                              1
Team Leader Pre-sale                                                      1
Pre-Sales Executive/Officer                                               1
Lead Associate - Training - Pune                                          1
Senior Consultant - Software Engineering                                  1
ServiceNow 4 ITSM and 2 CSM Consultant                                    1
Full Stack Developer (.Net Core Angular)                                  1
Tech Coord-Inv Ops- Trade Settlements                                     1
Team Leader Invst Ops Trade Settlements                                   1
Senior Analyst-Invst Ops-Trade Settlements                                1
Hiring For React.JS Developer - Immediate Joiner                          1
Outbound Sales Executive, Telecaller Voice Process                        1
Account executive (US Payroll)                                            1
Urgent Vacancy Telesales Executive, Telecaller                            1
Sr. Group Mgr.- Insurance Ops (Motor Claims- Subrogation- US Mkt) PUNE    1
Executive - Documentation (Ocean Exports)                                 1
Name: count, dtype: int64

In [15]:
df['company'].nunique()

15313

In [16]:
len(df['company'].unique())

15314

In [17]:
df['company'].value_counts()

company
Lavya Associates             5131
Accenture                    4446
Tech Mahindra                1285
Varite India Pvt. Ltd.        882
Hucon                         852
                             ... 
Dua Solutions                   1
Sheeba Medical Services         1
CONNECTING TALENT               1
Afcons Infotech                 1
People Connect Placements       1
Name: count, Length: 15313, dtype: int64

In [18]:
df['company'].value_counts().head(25)

company
Lavya Associates          5131
Accenture                 4446
Tech Mahindra             1285
Varite India Pvt. Ltd.     882
Hucon                      852
IBM                        656
Megma Services             525
Citiustech                 511
Infosys                    505
D'source                   502
RCPC                       449
Hexaware Technologies      431
Tekpillar Services         377
Mindtree                   368
Capgemini                  338
BAJAJ FINSERVE             317
Oracle                     304
Hdfc Bank                  300
Deloitte                   285
Radical Technologies       283
EY                         279
Marriott                   263
Red Hat                    259
HCLTech                    254
Wintax Solution            246
Name: count, dtype: int64

In [19]:
df['company'].duplicated().sum()

np.int64(63833)

In [20]:
df['experience']

0          1-6 Yrs
1        11-15 Yrs
2          3-7 Yrs
3          3-7 Yrs
4          5-7 Yrs
           ...    
79142      3-8 Yrs
79143    10-13 Yrs
79144     5-10 Yrs
79145      0-2 Yrs
79146     8-12 Yrs
Name: experience, Length: 79147, dtype: object

In [21]:
df['experience'].value_counts().head(10)

experience
5-10 Yrs    8041
3-8 Yrs     4451
1-6 Yrs     3302
4-9 Yrs     3186
2-7 Yrs     3079
0-5 Yrs     2825
2-5 Yrs     2778
1-3 Yrs     2706
2-4 Yrs     2488
4-6 Yrs     2435
Name: count, dtype: int64

In [22]:
df['experience'].value_counts().tail(10)

experience
16-19 Yrs    1
25-26 Yrs    1
21-27 Yrs    1
15-15 Yrs    1
22-25 Yrs    1
20-26 Yrs    1
22-28 Yrs    1
19-24 Yrs    1
16-21 Yrs    1
20-29 Yrs    1
Name: count, dtype: int64

In [23]:
df['experience'].duplicated().sum()

np.int64(78915)

In [24]:
df.columns

Index(['job_id', 'job_role', 'company', 'experience', 'salary', 'location',
       'rating', 'reviews', 'resposibilities', 'posted_on', 'job_link',
       'company_link'],
      dtype='object')

In [25]:
df['salary'].value_counts()

salary
Not disclosed                                          45544
6,50,000 - 9,00,000 PA.                                 3330
6,00,000 - 8,00,000 PA.                                 1114
5,00,000 - 10,00,000 PA.                                1087
2,50,000 - 3,50,000 PA.                                  599
                                                       ...  
4,00,000 - 9,00,000 PA. (Including Variable: 20.0%)        1
4,00,00,000 - 4,50,00,000 PA.                              1
2,00,000 - 6,00,000 PA. (Including Variable: 30.0%)        1
60,00,000 - 1,00,00,000 PA.                                1
75,00,000 - 1,00,00,000 PA.                                1
Name: count, Length: 813, dtype: int64

In [26]:
df['salary'].value_counts().head(10)

salary
Not disclosed               45544
6,50,000 - 9,00,000 PA.      3330
6,00,000 - 8,00,000 PA.      1114
5,00,000 - 10,00,000 PA.     1087
2,50,000 - 3,50,000 PA.       599
2,00,000 - 3,00,000 PA.       486
3,00,000 - 5,00,000 PA.       450
3,00,000 - 6,00,000 PA.       424
2,00,000 - 4,00,000 PA.       382
2,00,000 - 5,00,000 PA.       367
Name: count, dtype: int64

In [27]:
df['location'].value_counts()

location
Bangalore/Bengaluru                                                                          10035
Hyderabad/Secunderabad                                                                        3736
Pune                                                                                          3216
Chennai                                                                                       2953
Mumbai                                                                                        2818
                                                                                             ...  
Panvel, Andheri, Nerul                                                                           1
Surendranagar, Mehsana, Gandhinagar, Sanand, Ahmedabad, Rajkot                                   1
Kalol, Mahesana, Surendranagar, Mundra, Morvi, Gandhinagar, Patan, Deesa, Ahmedabad              1
Bhavnagar, Surendranagar, Mehsana, Gandhidham, Bhuj, Gandhinagar, Patan, Deesa, Ahmedabad        1
Kalol, Surendranagar, Mehsana, Ahmedabad, Porbandar, Rajkot                                      1
Name: count, Length: 9038, dtype: int64

In [28]:
df['location'].value_counts().head(20)

location
Bangalore/Bengaluru                                                                       10035
Hyderabad/Secunderabad                                                                     3736
Pune                                                                                       3216
Chennai                                                                                    2953
Mumbai                                                                                     2818
Gurgaon/Gurugram                                                                           2733
Noida                                                                                      1678
Permanent Remote                                                                           1644
Ahmedabad                                                                                  1227
Kolkata                                                                                     923
Delhi / NCR                                                                                 922
New Delhi                                                                                   900
Kolkata, Mumbai, Hyderabad/Secunderabad, Pune, Chennai, Bangalore/Bengaluru                 786
Kolkata, Mumbai, Hyderabad/Secunderabad, Pune, Chennai, Bangalore/Bengaluru, delhi ncr      772
Nasik/Nashik                                                                                593
Hybrid - Bangalore/Bengaluru                                                                549
Jaipur                                                                                      516
Coimbatore                                                                                  478
Vadodara                                                                                    462
Surat                                                                                       434
Name: count, dtype: int64

In [29]:
df['rating'].value_counts()

rating
4.1    7931
4.0    5239
4.2    4007
3.9    3995
3.7    3487
3.8    3089
4.3    1782
3.6    1617
4.4    1357
3.5    1333
5.0    1236
4.5    1134
3.3     825
4.8     763
3.4     668
4.6     598
4.7     535
3.2     526
3.1     449
2.8     335
3.0     319
2.9     259
4.9     208
2.5     181
2.7     165
2.6     145
1.0     122
2.1      99
2.3      77
2.2      68
2.4      67
2.0      55
1.6      48
1.8      43
1.7      40
1.5      36
1.1      29
1.9      28
1.4      20
1.3      20
1.2      13
Name: count, dtype: int64

In [30]:
df['rating'].isnull().sum()


np.int64(36199)

In [31]:
df['resposibilities'].value_counts().head(20)


resposibilities
agent,tied,business,agency,dst,channel,direct,team                                                                                                                      1800
tied,agency channel,bfsi,agency,team handling,advisors,training,branch management                                                                                       1508
Team Management,Tied Channel,Alternate Channel,Agency Channel,Life Insurance,Selling Skills,Team Training,team training manager                                          876
training development,partnership distribution,learning development,training,training manager,learning & development                                                      630
Life Insurance,BFSI,Bancassurance,team handling,Channel Management,bank branches,sales,banking                                                                           317
java                                                                                                                                                                     257
Itam,ITSM,Hrsd,GRC,Itom,CSD                                                                                                                                              254
Kafka,sql,spring boot,REST API,java,Lambda,Design Patterns,REST API's                                                                                                    254
Bengali,International BPO,BPO,Customer Service,Non Voice Process,English Language,Annotation                                                                             254
SAP BO,PL / SQL,Oracle SQL,SAP Business Objects 4 x ( Information Design Tool,Universe Design Tool,Web Intelligence,SAP BO Administration,SAP BO Dashboards              254
Informatica Mdm,etl,business intelligence,Agile,MDM,Informatica,BI Space,SAP BODI                                                                                        254
c#,.net core,javascript,Microservices,Angular,CSS,Asp Net,ASP NET MVC                                                                                                    254
Kubernetes Deployment,Kubernetes,Cloud Computing,GCP,Openshift,AWS,Microservices,Azure Devops                                                                            254
Unified Functional Testing,automation testing,VB SCRIPT,Uft Testing,Hp Alm,UFT,object - oriented automation,communication                                                254
Business process,Change management,Team management,Stakeholder management,Continuous improvement,Program Management,Data Analytics,Strategy consulting                   254
java,Spring,Microservices,GCP,communication,Azure,Kubernetes,Apigee                                                                                                      254
solution design,Component Design,Solution Architecture,Enterprise Telecom Product,Cpq,Product Modelling,TM Forum Frameworks,Fulfillment and Assurance Billing Design     254
order management,Warehouse Management,supply chain management,Analytics                                                                                                  254
Product management,Change management,Consulting,Machine learning,Python,VBA,Artificial Intelligence,Text analysis                                                        254
Business process,Consulting,Forensic,Stakeholder management,Analytics,Analytical skills,Service delivery,Audit compliance                                                254
Name: count, dtype: int64

In [32]:
df['posted_on'].unique()

array(['1 Day Ago', '3 Days Ago', '2 Days Ago', 'Walk-in3 Days Ago',
       '7 Days Ago', '9 Days Ago', '5 Days Ago', '4 Days Ago',
       '6 Days Ago', '8 Days Ago', 'Today', '10 Days Ago',
       'Walk-in1 Day Ago', 'Walk-in2 Days Ago', 'Walk-in4 Days Ago',
       'Walk-in5 Days Ago', 'Just Now', 'Walk-inJust Now',
       'Walk-in9 Days Ago', 'Walk-in8 Days Ago', 'Walk-in10 Days Ago',
       'Walk-in6 Days Ago', 'Few Hours Ago', 'Walk-in7 Days Ago',
       'Walk-inFew Hours Ago', 'Walk-inToday', nan], dtype=object)

## 1.1 **Checking Null Values**

In [33]:
df.isnull().sum()

job_id               480
job_role             480
company              481
experience          1749
salary               480
location            1706
rating             36199
reviews            36199
resposibilities      500
posted_on            480
job_link             480
company_link         480
dtype: int64

### 1.2 **Checking Duplicate Values**

In [34]:
df.duplicated().sum()

np.int64(5801)

In [35]:
df.duplicated('job_id').sum()

np.int64(6137)

### 1.3 **Removing Unwanted Columns**

In [36]:
del df['company_link']
del df['job_link']
del df['posted_on']

In [37]:
df.columns

Index(['job_id', 'job_role', 'company', 'experience', 'salary', 'location',
       'rating', 'reviews', 'resposibilities'],
      dtype='object')

## **2. Data Cleaning**

### **2.1 Dealing with Null Values**
- Remove rows with null values except rating & reviews
- Deal with rating and reviews separately

In [38]:
df.isnull().sum()

job_id               480
job_role             480
company              481
experience          1749
salary               480
location            1706
rating             36199
reviews            36199
resposibilities      500
dtype: int64

In [39]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 79147 entries, 0 to 79146
Data columns (total 9 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   job_id           78667 non-null  float64
 1   job_role         78667 non-null  object 
 2   company          78666 non-null  object 
 3   experience       77398 non-null  object 
 4   salary           78667 non-null  object 
 5   location         77441 non-null  object 
 6   rating           42948 non-null  float64
 7   reviews          42948 non-null  object 
 8   resposibilities  78647 non-null  object 
dtypes: float64(2), object(7)
memory usage: 5.4+ MB


In [40]:
df.dropna(subset = ['job_id'], inplace = True)
df

job_id                                           job_role  \
0      7.012301e+10      Branch Banking - Calling For Women Candidates   
1      6.012391e+10                       Product Owner Senior Manager   
2      6.012391e+10  Employee Relations and Policies Associate Manager   
3      6.012391e+10         Employee Relations and Policies Specialist   
4      6.012301e+10                                  SAP BO Consultant   
...             ...                                                ...   
79142  6.012301e+10                           .Net Fullstack Developer   
79143  1.122201e+10        Senior (BI) Business Intelligence Architect   
79144  1.512220e+11                                  PowerBI Developer   
79145  2.712220e+11                         Customer Service Associate   
79146  2.112220e+11                   Senior Java BED ( PAN India)- KG   

                     company experience                     salary  \
0                  Hdfc Bank    1-6 Yrs              Not disclosed   
1                  Accenture  11-15 Yrs              Not disclosed   
2                  Accenture    3-7 Yrs              Not disclosed   
3                  Accenture    3-7 Yrs              Not disclosed   
4                   Mindtree    5-7 Yrs              Not disclosed   
...                      ...        ...                        ...   
79142             Citiustech    3-8 Yrs   7,00,000 - 17,00,000 PA.   
79143  Hexaware Technologies  10-13 Yrs              Not disclosed   
79144                     EY   5-10 Yrs              Not disclosed   
79145          Tech Mahindra    0-2 Yrs    2,50,000 - 3,50,000 PA.   
79146          Tech Mahindra   8-12 Yrs  25,00,000 - 27,50,000 PA.   

                                                location  rating  \
0      Kolkata, Hyderabad/Secunderabad, Pune, Ahmedab...     4.0   
1      Kolkata, Mumbai, Hyderabad/Secunderabad, Pune,...     4.1   
2      Kolkata, Mumbai, Hyderabad/Secunderabad, Pune,...     4.1   
3      Kolkata, Mumbai, Hyderabad/Secunderabad, Pune,...     4.1   
4      Hybrid - Kolkata, Hyderabad/Secunderabad, Pune...     4.1   
...                                                  ...     ...   
79142  Hybrid - Pune, Bangalore/Bengaluru, Mumbai (Al...     4.0   
79143  Hybrid - Pune, Chennai, Bangalore/Bengaluru, M...     3.7   
79144                  Hybrid - Pune, Mumbai (All Areas)     3.8   
79145  Temp. WFH - Noida, Kolkata, New Delhi, Hyderab...     3.7   
79146  Temp. WFH - Noida, Kolkata, Hyderabad/Secunder...     3.7   

             reviews                                    resposibilities  
0      39110 Reviews     Customer Service,Sales,Relationship Management  
1      32129 Reviews  Product management,Market analysis,Change mana...  
2      32129 Reviews  Business process,Change management,Team manage...  
3      32129 Reviews  Business process,Change management,Team manage...  
4       3759 Reviews  SAP BO,PL / SQL,Oracle SQL,SAP Business Object...  
...              ...                                                ...  
79142    742 Reviews  c#,.net core,javascript,Microservices,Angular,...  
79143   3172 Reviews  Informatica Mdm,etl,business intelligence,Agil...  
79144   6267 Reviews                            pagination,Power Bi,Dax  
79145  21440 Reviews  Bengali,International BPO,BPO,Customer Service...  
79146  21440 Reviews  Kafka,sql,spring boot,REST API,java,Lambda,Des...  

[78667 rows x 9 columns]

In [41]:
79147-78667

480

In [42]:
df.isnull().sum()

job_id                 0
job_role               0
company                1
experience          1269
salary                 0
location            1226
rating             35719
reviews            35719
resposibilities       20
dtype: int64

In [43]:
df.dropna(subset = ['resposibilities'], inplace = True)
df.isnull().sum()

job_id                 0
job_role               0
company                1
experience          1268
salary                 0
location            1226
rating             35718
reviews            35718
resposibilities        0
dtype: int64

In [44]:
df.dropna(subset = ['company'], inplace = True)
df.isnull().sum()

job_id                 0
job_role               0
company                0
experience          1268
salary                 0
location            1226
rating             35718
reviews            35718
resposibilities        0
dtype: int64

In [45]:
df['experience']

0          1-6 Yrs
1        11-15 Yrs
2          3-7 Yrs
3          3-7 Yrs
4          5-7 Yrs
           ...    
79142      3-8 Yrs
79143    10-13 Yrs
79144     5-10 Yrs
79145      0-2 Yrs
79146     8-12 Yrs
Name: experience, Length: 78646, dtype: object

In [46]:
len(df['experience'].unique())

232

In [47]:
df['experience'].value_counts().head(20)

experience
5-10 Yrs    8039
3-8 Yrs     4450
1-6 Yrs     3302
4-9 Yrs     3186
2-7 Yrs     3077
0-5 Yrs     2825
2-5 Yrs     2776
1-3 Yrs     2704
2-4 Yrs     2488
4-6 Yrs     2435
3-5 Yrs     2392
0-2 Yrs     2140
1-5 Yrs     2044
3-7 Yrs     2022
0-1 Yrs     2000
1-4 Yrs     1735
7-12 Yrs    1679
2-6 Yrs     1678
3-6 Yrs     1624
0-3 Yrs     1396
Name: count, dtype: int64

In [48]:
df['experience'] = df['experience'].fillna('5-10 Yrs')

In [49]:
df.isnull().sum()

job_id                 0
job_role               0
company                0
experience             0
salary                 0
location            1226
rating             35718
reviews            35718
resposibilities        0
dtype: int64

In [50]:
df['location'].value_counts()

location
Bangalore/Bengaluru                                                                                            10027
Hyderabad/Secunderabad                                                                                          3736
Pune                                                                                                            3216
Chennai                                                                                                         2953
Mumbai                                                                                                          2818
                                                                                                               ...  
Udaipur, Lucknow, Jammu, Jodhpur                                                                                   1
Kolkata, Bareilly, Rudrapur, Haldwani, Jamshedpur, Nainital, Pilibhit                                              1
Kota, Ahmedabad, Chennai                                                                                           1
Guwahati, Kolkata, Chandigarh, Patna, North Twenty Four Parganas, Balasore, Ranchi, Delhi / NCR                    1
Hybrid - Bhubaneswar, Kolkata, Hyderabad/Secunderabad, Pune, Gurgaon/Gurugram, Chennai, Bangalore/Bengaluru        1
Name: count, Length: 9034, dtype: int64

In [51]:
df['location'] = df['location'].fillna('Bangalore/Bengaluru')

In [52]:
df.isnull().sum()

job_id                 0
job_role               0
company                0
experience             0
salary                 0
location               0
rating             35718
reviews            35718
resposibilities        0
dtype: int64

In [53]:
df['reviews'].value_counts()

reviews
32129 Reviews    2928
32151 Reviews    1518
2 Reviews        1457
21440 Reviews    1286
3 Reviews        1071
                 ... 
882 Reviews         1
430 Reviews         1
1305 Reviews        1
734 Reviews         1
883 Reviews         1
Name: count, Length: 980, dtype: int64

In [54]:
df['reviews'].unique()

array(['39110 Reviews', '32129 Reviews', '3759 Reviews', '10263 Reviews',
       '23786 Reviews', '3843 Reviews', '21440 Reviews', '742 Reviews',
       '18348 Reviews', '218 Reviews', '3172 Reviews', '6258 Reviews',
       '1441 Reviews', '33480 Reviews', '50163 Reviews', '19288 Reviews',
       '1190 Reviews', '28658 Reviews', '1686 Reviews', '981 Reviews',
       '24041 Reviews', '370 Reviews', '56 Reviews', '4744 Reviews',
       '601 Reviews', '2060 Reviews', '345 Reviews', '1138 Reviews',
       '1462 Reviews', '1503 Reviews', '3827 Reviews', '274 Reviews',
       '802 Reviews', '5169 Reviews', '311 Reviews', '4786 Reviews',
       '59 Reviews', '2611 Reviews', '321 Reviews', '1232 Reviews',
       '236 Reviews', '613 Reviews', '986 Reviews', '71 Reviews',
       '102 Reviews', '4795 Reviews', '3118 Reviews', '18335 Reviews',
       '1747 Reviews', '2239 Reviews', '136 Reviews', '234 Reviews',
       '1257 Reviews', '4825 Reviews', '351 Reviews', '706 Reviews',
       '242 Review

In [55]:
df['rating'].isnull().sum()

np.int64(35718)

In [56]:
df['rating']

0        4.0
1        4.1
2        4.1
3        4.1
4        4.1
        ... 
79142    4.0
79143    3.7
79144    3.8
79145    3.7
79146    3.7
Name: rating, Length: 78646, dtype: float64

In [57]:
df['rating'] = df['rating'].fillna(0.0)

In [58]:
df['reviews'] = df['reviews'].fillna('0 Reviews')

In [59]:
df.isnull().sum()

job_id             0
job_role           0
company            0
experience         0
salary             0
location           0
rating             0
reviews            0
resposibilities    0
dtype: int64

### **2.2 Dealing with Duplicate Values**

In [60]:
df.drop_duplicates(subset = 'job_id', inplace = True)

In [61]:
df

job_id                                           job_role  \
0      7.012301e+10      Branch Banking - Calling For Women Candidates   
1      6.012391e+10                       Product Owner Senior Manager   
2      6.012391e+10  Employee Relations and Policies Associate Manager   
3      6.012391e+10         Employee Relations and Policies Specialist   
4      6.012301e+10                                  SAP BO Consultant   
...             ...                                                ...   
73762  2.012300e+10            Partner Success Executive/Edtech/Punjab   
73763  2.012300e+10            Partner Success Associate/Edtech/Punjab   
73764  2.312220e+11  Hiring For International Voice Process | Gurga...   
73765  1.712200e+11                                   Fresher Engineer   
73766  7.102201e+10            Anatomy Trainer - Part Time / Full time   

                     company experience                   salary  \
0                  Hdfc Bank    1-6 Yrs            Not disclosed   
1                  Accenture  11-15 Yrs            Not disclosed   
2                  Accenture    3-7 Yrs            Not disclosed   
3                  Accenture    3-7 Yrs            Not disclosed   
4                   Mindtree    5-7 Yrs            Not disclosed   
...                      ...        ...                      ...   
73762       Parth Associates    1-5 Yrs  7,00,000 - 8,50,000 PA.   
73763       Parth Associates    1-5 Yrs  7,00,000 - 8,50,000 PA.   
73764   First Step Solutions    1-4 Yrs  3,00,000 - 4,50,000 PA.   
73765                 Sofcon    0-2 Yrs            Not disclosed   
73766  Indian Healthcare BPO    1-6 Yrs            Not disclosed   

                                                location  rating  \
0      Kolkata, Hyderabad/Secunderabad, Pune, Ahmedab...     4.0   
1      Kolkata, Mumbai, Hyderabad/Secunderabad, Pune,...     4.1   
2      Kolkata, Mumbai, Hyderabad/Secunderabad, Pune,...     4.1   
3      Kolkata, Mumbai, Hyderabad/Secunderabad, Pune,...     4.1   
4      Hybrid - Kolkata, Hyderabad/Secunderabad, Pune...     4.1   
...                                                  ...     ...   
73762                    Jalandhar, Chandigarh, Amritsar     0.0   
73763                            Ludhiana, Patiala, Moga     0.0   
73764                        New Delhi, Gurgaon/Gurugram     0.0   
73765  Kota, Udaipur, Banswara, Bhiwadi, Jaipur, Alwa...     2.8   
73766    Hybrid - Pondicherry/Puducherry, Salem, Chennai     0.0   

             reviews                                    resposibilities  
0      39110 Reviews     Customer Service,Sales,Relationship Management  
1      32129 Reviews  Product management,Market analysis,Change mana...  
2      32129 Reviews  Business process,Change management,Team manage...  
3      32129 Reviews  Business process,Change management,Team manage...  
4       3759 Reviews  SAP BO,PL / SQL,Oracle SQL,SAP Business Object...  
...              ...                                                ...  
73762      0 Reviews  CRM,Communication Skills,Presentation Skills,C...  
73763      0 Reviews  CRM,Communication Skills,Presentation Skills,C...  
73764      0 Reviews  Customer Service,US Process,Hospitality,Custom...  
73765      3 Reviews  ENGINEERING,B Tech Fresher,AutoCAD,Degree,PLC,...  
73766      0 Reviews  Lecturer Activities,Nursing Education,Trainer,...  

[72988 rows x 9 columns]

### **2.3 Change the Data Type of job_id column**

In [62]:
df['job_id'].astype(int)

0         70123006070
1         60123905908
2         60123905898
3         60123905897
4         60123008332
             ...     
73762     20123002989
73763     20123002957
73764    231222003986
73765    171220001449
73766     71022008732
Name: job_id, Length: 72988, dtype: int64

In [63]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 72988 entries, 0 to 73766
Data columns (total 9 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   job_id           72988 non-null  float64
 1   job_role         72988 non-null  object 
 2   company          72988 non-null  object 
 3   experience       72988 non-null  object 
 4   salary           72988 non-null  object 
 5   location         72988 non-null  object 
 6   rating           72988 non-null  float64
 7   reviews          72988 non-null  object 
 8   resposibilities  72988 non-null  object 
dtypes: float64(2), object(7)
memory usage: 5.6+ MB


### **2.4 Clean Experience Column**

In [64]:
df['experience'].unique()

array(['1-6 Yrs', '11-15 Yrs', '3-7 Yrs', '5-7 Yrs', '5-10 Yrs',
       '7-12 Yrs', '2-6 Yrs', '2-4 Yrs', '4-9 Yrs', '3-8 Yrs', '4-6 Yrs',
       '5-9 Yrs', '10-17 Yrs', '10-13 Yrs', '6-11 Yrs', '2-7 Yrs',
       '0-2 Yrs', '8-12 Yrs', '3-6 Yrs', '9-14 Yrs', '0-5 Yrs', '4-7 Yrs',
       '4-8 Yrs', '8-13 Yrs', '7-11 Yrs', '3-4 Yrs', '15-22 Yrs',
       '2-5 Yrs', '10-18 Yrs', '10-15 Yrs', '0-3 Yrs', '5-12 Yrs',
       '10-14 Yrs', '3-5 Yrs', '6-8 Yrs', '8-10 Yrs', '12-14 Yrs',
       '10-20 Yrs', '12-22 Yrs', '9-15 Yrs', '6-9 Yrs', '14-18 Yrs',
       '9-12 Yrs', '6-10 Yrs', '8-11 Yrs', '17-22 Yrs', '1-5 Yrs',
       '0-4 Yrs', '5-8 Yrs', '0-0 Yrs', '0-13 Yrs', '1-3 Yrs', '3-10 Yrs',
       '13-20 Yrs', '12-17 Yrs', '11-16 Yrs', '15-20 Yrs', '1-4 Yrs',
       '0-1 Yrs', '0-8 Yrs', '2-3 Yrs', '8-9 Yrs', '7-10 Yrs', '4-5 Yrs',
       '14-20 Yrs', '7-9 Yrs', '10-12 Yrs', '10-16 Yrs', '12-15 Yrs',
       '12-18 Yrs', '7-8 Yrs', '16-20 Yrs', '5-6 Yrs', '1-2 Yrs',
       '9-13 Yrs', '6-7 Yrs'

In [65]:
df['experience']

0          1-6 Yrs
1        11-15 Yrs
2          3-7 Yrs
3          3-7 Yrs
4          5-7 Yrs
           ...    
73762      1-5 Yrs
73763      1-5 Yrs
73764      1-4 Yrs
73765      0-2 Yrs
73766      1-6 Yrs
Name: experience, Length: 72988, dtype: object

In [66]:
df['experience'] =  df['experience'].str.replace('Yrs','')
df['experience']

0          1-6 
1        11-15 
2          3-7 
3          3-7 
4          5-7 
          ...  
73762      1-5 
73763      1-5 
73764      1-4 
73765      0-2 
73766      1-6 
Name: experience, Length: 72988, dtype: object

In [67]:
df['experience'] = df['experience'].str.split('-')
df['experience']

0          [1, 6 ]
1        [11, 15 ]
2          [3, 7 ]
3          [3, 7 ]
4          [5, 7 ]
           ...    
73762      [1, 5 ]
73763      [1, 5 ]
73764      [1, 4 ]
73765      [0, 2 ]
73766      [1, 6 ]
Name: experience, Length: 72988, dtype: object

In [68]:
df['min experience'] = df['experience'].str[0].str.strip().astype(int)
df['min experience']

0         1
1        11
2         3
3         3
4         5
         ..
73762     1
73763     1
73764     1
73765     0
73766     1
Name: min experience, Length: 72988, dtype: int64

In [69]:
df['max experience'] = df['experience'].str[1].str.strip().astype(int)
df['max experience']

0         6
1        15
2         7
3         7
4         7
         ..
73762     5
73763     5
73764     4
73765     2
73766     6
Name: max experience, Length: 72988, dtype: int64

In [70]:
df.columns

Index(['job_id', 'job_role', 'company', 'experience', 'salary', 'location',
       'rating', 'reviews', 'resposibilities', 'min experience',
       'max experience'],
      dtype='object')

In [71]:
df

job_id                                           job_role  \
0      7.012301e+10      Branch Banking - Calling For Women Candidates   
1      6.012391e+10                       Product Owner Senior Manager   
2      6.012391e+10  Employee Relations and Policies Associate Manager   
3      6.012391e+10         Employee Relations and Policies Specialist   
4      6.012301e+10                                  SAP BO Consultant   
...             ...                                                ...   
73762  2.012300e+10            Partner Success Executive/Edtech/Punjab   
73763  2.012300e+10            Partner Success Associate/Edtech/Punjab   
73764  2.312220e+11  Hiring For International Voice Process | Gurga...   
73765  1.712200e+11                                   Fresher Engineer   
73766  7.102201e+10            Anatomy Trainer - Part Time / Full time   

                     company experience                   salary  \
0                  Hdfc Bank    [1, 6 ]            Not disclosed   
1                  Accenture  [11, 15 ]            Not disclosed   
2                  Accenture    [3, 7 ]            Not disclosed   
3                  Accenture    [3, 7 ]            Not disclosed   
4                   Mindtree    [5, 7 ]            Not disclosed   
...                      ...        ...                      ...   
73762       Parth Associates    [1, 5 ]  7,00,000 - 8,50,000 PA.   
73763       Parth Associates    [1, 5 ]  7,00,000 - 8,50,000 PA.   
73764   First Step Solutions    [1, 4 ]  3,00,000 - 4,50,000 PA.   
73765                 Sofcon    [0, 2 ]            Not disclosed   
73766  Indian Healthcare BPO    [1, 6 ]            Not disclosed   

                                                location  rating  \
0      Kolkata, Hyderabad/Secunderabad, Pune, Ahmedab...     4.0   
1      Kolkata, Mumbai, Hyderabad/Secunderabad, Pune,...     4.1   
2      Kolkata, Mumbai, Hyderabad/Secunderabad, Pune,...     4.1   
3      Kolkata, Mumbai, Hyderabad/Secunderabad, Pune,...     4.1   
4      Hybrid - Kolkata, Hyderabad/Secunderabad, Pune...     4.1   
...                                                  ...     ...   
73762                    Jalandhar, Chandigarh, Amritsar     0.0   
73763                            Ludhiana, Patiala, Moga     0.0   
73764                        New Delhi, Gurgaon/Gurugram     0.0   
73765  Kota, Udaipur, Banswara, Bhiwadi, Jaipur, Alwa...     2.8   
73766    Hybrid - Pondicherry/Puducherry, Salem, Chennai     0.0   

             reviews                                    resposibilities  \
0      39110 Reviews     Customer Service,Sales,Relationship Management   
1      32129 Reviews  Product management,Market analysis,Change mana...   
2      32129 Reviews  Business process,Change management,Team manage...   
3      32129 Reviews  Business process,Change management,Team manage...   
4       3759 Reviews  SAP BO,PL / SQL,Oracle SQL,SAP Business Object...   
...              ...                                                ...   
73762      0 Reviews  CRM,Communication Skills,Presentation Skills,C...   
73763      0 Reviews  CRM,Communication Skills,Presentation Skills,C...   
73764      0 Reviews  Customer Service,US Process,Hospitality,Custom...   
73765      3 Reviews  ENGINEERING,B Tech Fresher,AutoCAD,Degree,PLC,...   
73766      0 Reviews  Lecturer Activities,Nursing Education,Trainer,...   

       min experience  max experience  
0                   1               6  
1                  11              15  
2                   3               7  
3                   3               7  
4                   5               7  
...               ...             ...  
73762               1               5  
73763               1               5  
73764               1               4  
73765               0               2  
73766               1               6  

[72988 rows x 11 columns]

In [72]:
del df['experience']
df

job_id                                           job_role  \
0      7.012301e+10      Branch Banking - Calling For Women Candidates   
1      6.012391e+10                       Product Owner Senior Manager   
2      6.012391e+10  Employee Relations and Policies Associate Manager   
3      6.012391e+10         Employee Relations and Policies Specialist   
4      6.012301e+10                                  SAP BO Consultant   
...             ...                                                ...   
73762  2.012300e+10            Partner Success Executive/Edtech/Punjab   
73763  2.012300e+10            Partner Success Associate/Edtech/Punjab   
73764  2.312220e+11  Hiring For International Voice Process | Gurga...   
73765  1.712200e+11                                   Fresher Engineer   
73766  7.102201e+10            Anatomy Trainer - Part Time / Full time   

                     company                   salary  \
0                  Hdfc Bank            Not disclosed   
1                  Accenture            Not disclosed   
2                  Accenture            Not disclosed   
3                  Accenture            Not disclosed   
4                   Mindtree            Not disclosed   
...                      ...                      ...   
73762       Parth Associates  7,00,000 - 8,50,000 PA.   
73763       Parth Associates  7,00,000 - 8,50,000 PA.   
73764   First Step Solutions  3,00,000 - 4,50,000 PA.   
73765                 Sofcon            Not disclosed   
73766  Indian Healthcare BPO            Not disclosed   

                                                location  rating  \
0      Kolkata, Hyderabad/Secunderabad, Pune, Ahmedab...     4.0   
1      Kolkata, Mumbai, Hyderabad/Secunderabad, Pune,...     4.1   
2      Kolkata, Mumbai, Hyderabad/Secunderabad, Pune,...     4.1   
3      Kolkata, Mumbai, Hyderabad/Secunderabad, Pune,...     4.1   
4      Hybrid - Kolkata, Hyderabad/Secunderabad, Pune...     4.1   
...                                                  ...     ...   
73762                    Jalandhar, Chandigarh, Amritsar     0.0   
73763                            Ludhiana, Patiala, Moga     0.0   
73764                        New Delhi, Gurgaon/Gurugram     0.0   
73765  Kota, Udaipur, Banswara, Bhiwadi, Jaipur, Alwa...     2.8   
73766    Hybrid - Pondicherry/Puducherry, Salem, Chennai     0.0   

             reviews                                    resposibilities  \
0      39110 Reviews     Customer Service,Sales,Relationship Management   
1      32129 Reviews  Product management,Market analysis,Change mana...   
2      32129 Reviews  Business process,Change management,Team manage...   
3      32129 Reviews  Business process,Change management,Team manage...   
4       3759 Reviews  SAP BO,PL / SQL,Oracle SQL,SAP Business Object...   
...              ...                                                ...   
73762      0 Reviews  CRM,Communication Skills,Presentation Skills,C...   
73763      0 Reviews  CRM,Communication Skills,Presentation Skills,C...   
73764      0 Reviews  Customer Service,US Process,Hospitality,Custom...   
73765      3 Reviews  ENGINEERING,B Tech Fresher,AutoCAD,Degree,PLC,...   
73766      0 Reviews  Lecturer Activities,Nursing Education,Trainer,...   

       min experience  max experience  
0                   1               6  
1                  11              15  
2                   3               7  
3                   3               7  
4                   5               7  
...               ...             ...  
73762               1               5  
73763               1               5  
73764               1               4  
73765               0               2  
73766               1               6  

[72988 rows x 10 columns]

### **2.5 Clean Salary Column**

In [73]:
df['salary']

0                  Not disclosed
1                  Not disclosed
2                  Not disclosed
3                  Not disclosed
4                  Not disclosed
                  ...           
73762    7,00,000 - 8,50,000 PA.
73763    7,00,000 - 8,50,000 PA.
73764    3,00,000 - 4,50,000 PA.
73765              Not disclosed
73766              Not disclosed
Name: salary, Length: 72988, dtype: object

In [74]:
df['salary'].nunique()

813

In [75]:
df['salary'].unique()

array(['Not disclosed', '7,00,000 - 17,00,000 PA.',
       '10,00,000 - 20,00,000 PA.',
       '4,00,000 - 9,00,000 PA. (Including Variable: 20.0%)',
       '2,50,000 - 3,50,000 PA.', '25,00,000 - 27,50,000 PA.',
       '13,00,000 - 22,50,000 PA.', '8,50,000 - 18,50,000 PA.',
       '4,25,000 - 4,50,000 PA.', '5,50,000 - 15,50,000 PA.',
       '5,50,000 - 14,00,000 PA.', '8,00,000 - 18,00,000 PA.',
       '50,000 - 3,00,000 PA.', '4,75,000 PA.', '6,00,000 - 8,00,000 PA.',
       '8,00,000 - 12,00,000 PA.', '5,00,000 - 15,00,000 PA.',
       '6,00,000 - 16,00,000 PA.', '5,00,000 - 14,00,000 PA.',
       '10,00,000 - 18,00,000 PA.', '18,00,000 - 20,00,000 PA.',
       '4,00,000 - 7,00,000 PA.', '2,00,000 - 7,00,000 PA.',
       '20,00,000 - 30,00,000 PA.', '8,00,000 - 16,00,000 PA.',
       '3,00,000 - 6,50,000 PA.', '2,00,000 - 6,00,000 PA.', '50,000 PA.',
       '15,00,000 - 20,00,000 PA.', '9,00,000 - 19,00,000 PA.',
       '9,00,000 - 17,00,000 PA.', '22,50,000 - 25,00,000 PA.',
    

In [76]:
df['salary'].nunique()

813

In [77]:
df['salary'].value_counts()

salary
Not disclosed                                            40927
6,50,000 - 9,00,000 PA.                                   3325
6,00,000 - 8,00,000 PA.                                   1109
5,00,000 - 10,00,000 PA.                                  1087
2,00,000 - 3,00,000 PA.                                    478
                                                         ...  
3,50,000 - 5,50,000 PA. (Including Variable: 25.0%)          1
7,50,00,000 - 9.5 Cr and above PA.                           1
37,50,000 - 70,00,000 PA.                                    1
5,00,000 - 12,00,000 PA. (Including Variable: 100.0%)        1
75,00,000 - 1,00,00,000 PA.                                  1
Name: count, Length: 813, dtype: int64

In [78]:
df

job_id                                           job_role  \
0      7.012301e+10      Branch Banking - Calling For Women Candidates   
1      6.012391e+10                       Product Owner Senior Manager   
2      6.012391e+10  Employee Relations and Policies Associate Manager   
3      6.012391e+10         Employee Relations and Policies Specialist   
4      6.012301e+10                                  SAP BO Consultant   
...             ...                                                ...   
73762  2.012300e+10            Partner Success Executive/Edtech/Punjab   
73763  2.012300e+10            Partner Success Associate/Edtech/Punjab   
73764  2.312220e+11  Hiring For International Voice Process | Gurga...   
73765  1.712200e+11                                   Fresher Engineer   
73766  7.102201e+10            Anatomy Trainer - Part Time / Full time   

                     company                   salary  \
0                  Hdfc Bank            Not disclosed   
1                  Accenture            Not disclosed   
2                  Accenture            Not disclosed   
3                  Accenture            Not disclosed   
4                   Mindtree            Not disclosed   
...                      ...                      ...   
73762       Parth Associates  7,00,000 - 8,50,000 PA.   
73763       Parth Associates  7,00,000 - 8,50,000 PA.   
73764   First Step Solutions  3,00,000 - 4,50,000 PA.   
73765                 Sofcon            Not disclosed   
73766  Indian Healthcare BPO            Not disclosed   

                                                location  rating  \
0      Kolkata, Hyderabad/Secunderabad, Pune, Ahmedab...     4.0   
1      Kolkata, Mumbai, Hyderabad/Secunderabad, Pune,...     4.1   
2      Kolkata, Mumbai, Hyderabad/Secunderabad, Pune,...     4.1   
3      Kolkata, Mumbai, Hyderabad/Secunderabad, Pune,...     4.1   
4      Hybrid - Kolkata, Hyderabad/Secunderabad, Pune...     4.1   
...                                                  ...     ...   
73762                    Jalandhar, Chandigarh, Amritsar     0.0   
73763                            Ludhiana, Patiala, Moga     0.0   
73764                        New Delhi, Gurgaon/Gurugram     0.0   
73765  Kota, Udaipur, Banswara, Bhiwadi, Jaipur, Alwa...     2.8   
73766    Hybrid - Pondicherry/Puducherry, Salem, Chennai     0.0   

             reviews                                    resposibilities  \
0      39110 Reviews     Customer Service,Sales,Relationship Management   
1      32129 Reviews  Product management,Market analysis,Change mana...   
2      32129 Reviews  Business process,Change management,Team manage...   
3      32129 Reviews  Business process,Change management,Team manage...   
4       3759 Reviews  SAP BO,PL / SQL,Oracle SQL,SAP Business Object...   
...              ...                                                ...   
73762      0 Reviews  CRM,Communication Skills,Presentation Skills,C...   
73763      0 Reviews  CRM,Communication Skills,Presentation Skills,C...   
73764      0 Reviews  Customer Service,US Process,Hospitality,Custom...   
73765      3 Reviews  ENGINEERING,B Tech Fresher,AutoCAD,Degree,PLC,...   
73766      0 Reviews  Lecturer Activities,Nursing Education,Trainer,...   

       min experience  max experience  
0                   1               6  
1                  11              15  
2                   3               7  
3                   3               7  
4                   5               7  
...               ...             ...  
73762               1               5  
73763               1               5  
73764               1               4  
73765               0               2  
73766               1               6  

[72988 rows x 10 columns]

In [79]:
df['salary'].unique()

array(['Not disclosed', '7,00,000 - 17,00,000 PA.',
       '10,00,000 - 20,00,000 PA.',
       '4,00,000 - 9,00,000 PA. (Including Variable: 20.0%)',
       '2,50,000 - 3,50,000 PA.', '25,00,000 - 27,50,000 PA.',
       '13,00,000 - 22,50,000 PA.', '8,50,000 - 18,50,000 PA.',
       '4,25,000 - 4,50,000 PA.', '5,50,000 - 15,50,000 PA.',
       '5,50,000 - 14,00,000 PA.', '8,00,000 - 18,00,000 PA.',
       '50,000 - 3,00,000 PA.', '4,75,000 PA.', '6,00,000 - 8,00,000 PA.',
       '8,00,000 - 12,00,000 PA.', '5,00,000 - 15,00,000 PA.',
       '6,00,000 - 16,00,000 PA.', '5,00,000 - 14,00,000 PA.',
       '10,00,000 - 18,00,000 PA.', '18,00,000 - 20,00,000 PA.',
       '4,00,000 - 7,00,000 PA.', '2,00,000 - 7,00,000 PA.',
       '20,00,000 - 30,00,000 PA.', '8,00,000 - 16,00,000 PA.',
       '3,00,000 - 6,50,000 PA.', '2,00,000 - 6,00,000 PA.', '50,000 PA.',
       '15,00,000 - 20,00,000 PA.', '9,00,000 - 19,00,000 PA.',
       '9,00,000 - 17,00,000 PA.', '22,50,000 - 25,00,000 PA.',
    

In [133]:
df['salary'] = df['salary'].str.replace("PA.", "").str.replace("Less than 50000-Less than 50000", "Not disclosed - Not disclosed").str.replace("Any Postgraduate Any Graduate-Any Postgraduate Any Graduate","Not disclosed - Not disclosed").str.replace("Graduation Not Required-Graduation Not Required","Not disclosed - Not disclosed").str.replace("M.Tech MCA MS/M.Sc-M.Tech MCA MS/M.Sc","Not disclosed - Not disclosed").str.replace("CA MBA/PGDM M.Com B.Com-CA MBA/PGDM M.Com B.Com","Not disclosed - Not disclosed").str.replace("MBA/PGDM-MBA/PGDM","Not disclosed - Not disclosed").str.replace("B.Tech/B.E. BCA-B.Tech/B.E. BCA","Not disclosed - Not disclosed").str.replace("MS/M.Sc-MS/M.Sc","Not disclosed - Not disclosed").str.replace("MBBS-MBBS","Not disclosed - Not disclosed").str.replace("MCA B.Tech/B.E.-MCA B.Tech/B.E.","Not disclosed - Not disclosed").str.replace("90000000 - 9.5 Cr and above","9,00,00,000-9,50,00,000 ").str.replace("60000000 - 9.5 Cr and above","6,00,00,000 -9,50,00,000").str.replace("Less than 5000 - 5000","Not disclosed - 5,000").str.replace("Less than 50000 - 100000","Not disclosed - 1,00,000").str.replace("75000000 - 9.5 Cr and above","7,50,00,000 - 9,50,00,000 ").str.replace("9.5 Cr and above","9,50,00,000 - Not disclosed").str.replace("7,00,00,000 - 9.5 Cr and above ","7,00,00,000 - Not disclosed").str.replace("Any Graduate", "Not disclosed - Not disclosed").str.replace('Any Postgraduate, Any Graduate','Not disclosed - Not disclosed').str.replace("Graduation Not Required", "Not disclosed - Not disclosed").str.replace('M.Tech, MCA, MS/M.Sc(Science), B.Tech/B.E.',"Not disclosed - Not disclosed").str.replace('MCA, B.Tech/B.E.' , "Not disclosed - Not disclosed" ).str.replace('MS/M.Sc(Science), M.A', "Not disclosed - Not disclosed").str.replace('B.Tech/B.E., BCA', "Not disclosed - Not disclosed").str.replace("MBBS","Not disclosed - Not disclosed").str.replace("MBA/PGDM","Not disclosed - Not disclosed").str.replace("CA, MBA/PGDM, M.Com, B.Com", "Not disclosed - Not disclosed").str.replace('Any Postgraduate, 0-0', "Not disclosed - Not disclosed").str.replace("Less than 50,000 ", "Not disclosed - 50,000").str.replace('CA, 0-0, M.Com, B.Com', "Not disclosed - Not disclosed").str.replace('MS/M.Sc(Science), 0-0, M.A', "Not disclosed - Not disclosed")

In [145]:
df["salary"].unique()

array(['Not disclosed', '7,00,000 - 17,00,000 ', '10,00,000 - 20,00,000 ',
       '4,00,000 - 9,00,000  (Including Variable: 20.0%)',
       '2,50,000 - 3,50,000 ', '25,00,000 - 27,50,000 ',
       '13,00,000 - 22,50,000 ', '8,50,000 - 18,50,000 ',
       '4,25,000 - 4,50,000 ', '5,50,000 - 15,50,000 ',
       '5,50,000 - 14,00,000 ', '8,00,000 - 18,00,000 ',
       '50,000 - 3,00,000 ', '4,75,000 ', '6,00,000 - 8,00,000 ',
       '8,00,000 - 12,00,000 ', '5,00,000 - 15,00,000 ',
       '6,00,000 - 16,00,000 ', '5,00,000 - 14,00,000 ',
       '10,00,000 - 18,00,000 ', '18,00,000 - 20,00,000 ',
       '4,00,000 - 7,00,000 ', '2,00,000 - 7,00,000 ',
       '20,00,000 - 30,00,000 ', '8,00,000 - 16,00,000 ',
       '3,00,000 - 6,50,000 ', '2,00,000 - 6,00,000 ', '50,000 ',
       '15,00,000 - 20,00,000 ', '9,00,000 - 19,00,000 ',
       '9,00,000 - 17,00,000 ', '22,50,000 - 25,00,000 ',
       '7,00,000 - 15,00,000 ', '20,00,000 - 35,00,000 ',
       '9,50,00,000 - 9,50,00,000  ', '5,00,00

In [146]:
df['min_salary'] = df['salary'].str.split("(").str[0].str.split('-').str[0]

In [147]:
df['max_salary'] = df['salary'].str.split("(").str[0].str.split('-').str[-1]

In [162]:
df['min_salary']

0        Not disclosed
1        Not disclosed
2        Not disclosed
3        Not disclosed
4        Not disclosed
             ...      
73762           700000
73763           700000
73764           300000
73765    Not disclosed
73766    Not disclosed
Name: min_salary, Length: 72988, dtype: object

In [163]:
df['min_salary'].unique()

array(['Not disclosed', '700000', '1000000', '400000', '250000',
       '2500000', '1300000', '850000', '425000', '550000', '800000',
       '50000', '475000', '600000', '500000', '1800000', '200000',
       '2000000', '300000', '1500000', '900000', '2250000', '95000000',
       '950000', '1200000', '100000', '350000', '175000', '225000',
       '650000', '325000', '375000', '1400000', '150000', '60000',
       '3000000', '1700000', '450000', '750000', '4000000', '125000',
       '3500000', '275000', '50000000', '1600000', '0', '80000', '10000',
       '1100000', '40000000', '90000', '2750000', '70000000', '1900000',
       '5500000', '35000000', '15000', '70000', '20000000', '85000000',
       '10000000', '80000000', '6000000', '15000000', '3750000',
       '65000000', '5000000', '60000000', '27500000', '30000000',
       '4500000', '90000000', '3250000', '8000000', '9000000', '17500000',
       '42500000', '7000000', '5000', '32500000', '7500000', '75000000',
       '22500000', '3750

In [164]:
df['min_salary'] = df['min_salary'].str.strip()

In [165]:
df['min_salary'].unique()

array(['Not disclosed', '700000', '1000000', '400000', '250000',
       '2500000', '1300000', '850000', '425000', '550000', '800000',
       '50000', '475000', '600000', '500000', '1800000', '200000',
       '2000000', '300000', '1500000', '900000', '2250000', '95000000',
       '950000', '1200000', '100000', '350000', '175000', '225000',
       '650000', '325000', '375000', '1400000', '150000', '60000',
       '3000000', '1700000', '450000', '750000', '4000000', '125000',
       '3500000', '275000', '50000000', '1600000', '0', '80000', '10000',
       '1100000', '40000000', '90000', '2750000', '70000000', '1900000',
       '5500000', '35000000', '15000', '70000', '20000000', '85000000',
       '10000000', '80000000', '6000000', '15000000', '3750000',
       '65000000', '5000000', '60000000', '27500000', '30000000',
       '4500000', '90000000', '3250000', '8000000', '9000000', '17500000',
       '42500000', '7000000', '5000', '32500000', '7500000', '75000000',
       '22500000', '3750

In [166]:
df['min_salary'] = df['min_salary'].str.replace(",", "")

In [167]:
df['min_salary'].unique()

array(['Not disclosed', '700000', '1000000', '400000', '250000',
       '2500000', '1300000', '850000', '425000', '550000', '800000',
       '50000', '475000', '600000', '500000', '1800000', '200000',
       '2000000', '300000', '1500000', '900000', '2250000', '95000000',
       '950000', '1200000', '100000', '350000', '175000', '225000',
       '650000', '325000', '375000', '1400000', '150000', '60000',
       '3000000', '1700000', '450000', '750000', '4000000', '125000',
       '3500000', '275000', '50000000', '1600000', '0', '80000', '10000',
       '1100000', '40000000', '90000', '2750000', '70000000', '1900000',
       '5500000', '35000000', '15000', '70000', '20000000', '85000000',
       '10000000', '80000000', '6000000', '15000000', '3750000',
       '65000000', '5000000', '60000000', '27500000', '30000000',
       '4500000', '90000000', '3250000', '8000000', '9000000', '17500000',
       '42500000', '7000000', '5000', '32500000', '7500000', '75000000',
       '22500000', '3750

In [168]:
df['min_salary'] = df['min_salary'].str.replace('Less than 5000', 'Not disclosed')

In [169]:
df['min_salary'].value_counts()

min_salary
Not disclosed    40928
650000            3424
200000            2915
300000            2831
500000            2309
                 ...  
42500000             1
32500000             1
75000000             1
22500000             1
37500000             1
Name: count, Length: 84, dtype: int64

In [170]:
df['max_salary'].unique()

array(['Not disclosed', ' 17,00,000 ', ' 20,00,000 ', ' 9,00,000  ',
       ' 3,50,000 ', ' 27,50,000 ', ' 22,50,000 ', ' 18,50,000 ',
       ' 4,50,000 ', ' 15,50,000 ', ' 14,00,000 ', ' 18,00,000 ',
       ' 3,00,000 ', '4,75,000 ', ' 8,00,000 ', ' 12,00,000 ',
       ' 15,00,000 ', ' 16,00,000 ', ' 7,00,000 ', ' 30,00,000 ',
       ' 6,50,000 ', ' 6,00,000 ', '50,000 ', ' 19,00,000 ',
       ' 25,00,000 ', ' 35,00,000 ', ' 9,50,00,000  ', ' 8,50,000 ',
       ' 13,00,000 ', ' 90,000 ', ' 4,75,000 ', ' 9,00,000 ',
       ' 7,50,000 ', ' 5,00,000 ', ' 4,25,000 ', ' 2,50,000 ',
       ' 16,50,000 ', ' 4,00,000 ', ' 70,000 ', ' 10,00,000 ',
       ' 3,25,000 ', ' 2,00,000 ', ' 40,00,000 ', ' 4,75,000  ',
       ' 5,50,000 ', ' 6,00,000  ', ' 7,00,000  ', ' 1,50,000 ',
       ' 2,75,000 ', ' 24,00,000 ', '3,00,000 ', '2,50,000 ',
       ' 11,00,000 ', ' 3,75,000 ', ' 45,00,000 ', ' 6,75,000 ',
       ' 22,00,000 ', '2,00,000 ', ' 2,25,000 ', ' 23,00,000 ',
       ' 50,00,000 ', '3,25,000

In [171]:
df['max_salary'] = df['max_salary'].str.strip()

In [172]:
df['max_salary'].unique()

array(['Not disclosed', '17,00,000', '20,00,000', '9,00,000', '3,50,000',
       '27,50,000', '22,50,000', '18,50,000', '4,50,000', '15,50,000',
       '14,00,000', '18,00,000', '3,00,000', '4,75,000', '8,00,000',
       '12,00,000', '15,00,000', '16,00,000', '7,00,000', '30,00,000',
       '6,50,000', '6,00,000', '50,000', '19,00,000', '25,00,000',
       '35,00,000', '9,50,00,000', '8,50,000', '13,00,000', '90,000',
       '7,50,000', '5,00,000', '4,25,000', '2,50,000', '16,50,000',
       '4,00,000', '70,000', '10,00,000', '3,25,000', '2,00,000',
       '40,00,000', '5,50,000', '1,50,000', '2,75,000', '24,00,000',
       '11,00,000', '3,75,000', '45,00,000', '6,75,000', '22,00,000',
       '2,25,000', '23,00,000', '50,00,000', '9,75,000', '32,50,000',
       '65,00,000', '1,75,000', '33,00,000', '0', '9,50,000', '31,00,000',
       '8,25,000', '1,00,000', '80,000', '15,000', '6,00,00,000',
       '1,25,000', '42,50,000', '7,25,000', '60,000', '21,00,000',
       '5,00,00,000', '9,00

In [173]:
df['max_salary'] = df['max_salary'].str.replace(",", "").replace('0', 'Not disclosed')
df['max_salary'].unique()

array(['Not disclosed', '1700000', '2000000', '900000', '350000',
       '2750000', '2250000', '1850000', '450000', '1550000', '1400000',
       '1800000', '300000', '475000', '800000', '1200000', '1500000',
       '1600000', '700000', '3000000', '650000', '600000', '50000',
       '1900000', '2500000', '3500000', '95000000', '850000', '1300000',
       '90000', '750000', '500000', '425000', '250000', '1650000',
       '400000', '70000', '1000000', '325000', '200000', '4000000',
       '550000', '150000', '275000', '2400000', '1100000', '375000',
       '4500000', '675000', '2200000', '225000', '2300000', '5000000',
       '975000', '3250000', '6500000', '175000', '3300000', '950000',
       '3100000', '825000', '100000', '80000', '15000', '60000000',
       '125000', '4250000', '725000', '60000', '2100000', '50000000',
       '90000000', '9000000', '1750000', '37500000', '25000', '3200000',
       '7000000', '1950000', '6000000', '625000', '30000000', '925000',
       '3750000', '8000

In [175]:
df['salary']

0               Not disclosed
1               Not disclosed
2               Not disclosed
3               Not disclosed
4               Not disclosed
                 ...         
73762    7,00,000 - 8,50,000 
73763    7,00,000 - 8,50,000 
73764    3,00,000 - 4,50,000 
73765           Not disclosed
73766           Not disclosed
Name: salary, Length: 72988, dtype: object

In [178]:
df

job_id                                           job_role  \
0      7.012301e+10      Branch Banking - Calling For Women Candidates   
1      6.012391e+10                       Product Owner Senior Manager   
2      6.012391e+10  Employee Relations and Policies Associate Manager   
3      6.012391e+10         Employee Relations and Policies Specialist   
4      6.012301e+10                                  SAP BO Consultant   
...             ...                                                ...   
73762  2.012300e+10            Partner Success Executive/Edtech/Punjab   
73763  2.012300e+10            Partner Success Associate/Edtech/Punjab   
73764  2.312220e+11  Hiring For International Voice Process | Gurga...   
73765  1.712200e+11                                   Fresher Engineer   
73766  7.102201e+10            Anatomy Trainer - Part Time / Full time   

                     company                salary  \
0                  Hdfc Bank         Not disclosed   
1                  Accenture         Not disclosed   
2                  Accenture         Not disclosed   
3                  Accenture         Not disclosed   
4                   Mindtree         Not disclosed   
...                      ...                   ...   
73762       Parth Associates  7,00,000 - 8,50,000    
73763       Parth Associates  7,00,000 - 8,50,000    
73764   First Step Solutions  3,00,000 - 4,50,000    
73765                 Sofcon         Not disclosed   
73766  Indian Healthcare BPO         Not disclosed   

                                                location  rating  \
0      Kolkata, Hyderabad/Secunderabad, Pune, Ahmedab...     4.0   
1      Kolkata, Mumbai, Hyderabad/Secunderabad, Pune,...     4.1   
2      Kolkata, Mumbai, Hyderabad/Secunderabad, Pune,...     4.1   
3      Kolkata, Mumbai, Hyderabad/Secunderabad, Pune,...     4.1   
4      Hybrid - Kolkata, Hyderabad/Secunderabad, Pune...     4.1   
...                                                  ...     ...   
73762                    Jalandhar, Chandigarh, Amritsar     0.0   
73763                            Ludhiana, Patiala, Moga     0.0   
73764                        New Delhi, Gurgaon/Gurugram     0.0   
73765  Kota, Udaipur, Banswara, Bhiwadi, Jaipur, Alwa...     2.8   
73766    Hybrid - Pondicherry/Puducherry, Salem, Chennai     0.0   

             reviews                                    resposibilities  \
0      39110 Reviews     Customer Service,Sales,Relationship Management   
1      32129 Reviews  Product management,Market analysis,Change mana...   
2      32129 Reviews  Business process,Change management,Team manage...   
3      32129 Reviews  Business process,Change management,Team manage...   
4       3759 Reviews  SAP BO,PL / SQL,Oracle SQL,SAP Business Object...   
...              ...                                                ...   
73762      0 Reviews  CRM,Communication Skills,Presentation Skills,C...   
73763      0 Reviews  CRM,Communication Skills,Presentation Skills,C...   
73764      0 Reviews  Customer Service,US Process,Hospitality,Custom...   
73765      3 Reviews  ENGINEERING,B Tech Fresher,AutoCAD,Degree,PLC,...   
73766      0 Reviews  Lecturer Activities,Nursing Education,Trainer,...   

       min experience  max experience     min_salary     max_salary  
0                   1               6  Not disclosed  Not disclosed  
1                  11              15  Not disclosed  Not disclosed  
2                   3               7  Not disclosed  Not disclosed  
3                   3               7  Not disclosed  Not disclosed  
4                   5               7  Not disclosed  Not disclosed  
...               ...             ...            ...            ...  
73762               1               5         700000         850000  
73763               1               5         700000         850000  
73764               1               4         300000         450000  
73765               0               2  Not disclosed  Not

In [179]:
df['max_salary']

0        Not disclosed
1        Not disclosed
2        Not disclosed
3        Not disclosed
4        Not disclosed
             ...      
73762           850000
73763           850000
73764           450000
73765    Not disclosed
73766    Not disclosed
Name: max_salary, Length: 72988, dtype: object

In [180]:
df['min_salary']

0        Not disclosed
1        Not disclosed
2        Not disclosed
3        Not disclosed
4        Not disclosed
             ...      
73762           700000
73763           700000
73764           300000
73765    Not disclosed
73766    Not disclosed
Name: min_salary, Length: 72988, dtype: object

In [181]:
del df['salary']

In [182]:
df

job_id                                           job_role  \
0      7.012301e+10      Branch Banking - Calling For Women Candidates   
1      6.012391e+10                       Product Owner Senior Manager   
2      6.012391e+10  Employee Relations and Policies Associate Manager   
3      6.012391e+10         Employee Relations and Policies Specialist   
4      6.012301e+10                                  SAP BO Consultant   
...             ...                                                ...   
73762  2.012300e+10            Partner Success Executive/Edtech/Punjab   
73763  2.012300e+10            Partner Success Associate/Edtech/Punjab   
73764  2.312220e+11  Hiring For International Voice Process | Gurga...   
73765  1.712200e+11                                   Fresher Engineer   
73766  7.102201e+10            Anatomy Trainer - Part Time / Full time   

                     company  \
0                  Hdfc Bank   
1                  Accenture   
2                  Accenture   
3                  Accenture   
4                   Mindtree   
...                      ...   
73762       Parth Associates   
73763       Parth Associates   
73764   First Step Solutions   
73765                 Sofcon   
73766  Indian Healthcare BPO   

                                                location  rating  \
0      Kolkata, Hyderabad/Secunderabad, Pune, Ahmedab...     4.0   
1      Kolkata, Mumbai, Hyderabad/Secunderabad, Pune,...     4.1   
2      Kolkata, Mumbai, Hyderabad/Secunderabad, Pune,...     4.1   
3      Kolkata, Mumbai, Hyderabad/Secunderabad, Pune,...     4.1   
4      Hybrid - Kolkata, Hyderabad/Secunderabad, Pune...     4.1   
...                                                  ...     ...   
73762                    Jalandhar, Chandigarh, Amritsar     0.0   
73763                            Ludhiana, Patiala, Moga     0.0   
73764                        New Delhi, Gurgaon/Gurugram     0.0   
73765  Kota, Udaipur, Banswara, Bhiwadi, Jaipur, Alwa...     2.8   
73766    Hybrid - Pondicherry/Puducherry, Salem, Chennai     0.0   

             reviews                                    resposibilities  \
0      39110 Reviews     Customer Service,Sales,Relationship Management   
1      32129 Reviews  Product management,Market analysis,Change mana...   
2      32129 Reviews  Business process,Change management,Team manage...   
3      32129 Reviews  Business process,Change management,Team manage...   
4       3759 Reviews  SAP BO,PL / SQL,Oracle SQL,SAP Business Object...   
...              ...                                                ...   
73762      0 Reviews  CRM,Communication Skills,Presentation Skills,C...   
73763      0 Reviews  CRM,Communication Skills,Presentation Skills,C...   
73764      0 Reviews  Customer Service,US Process,Hospitality,Custom...   
73765      3 Reviews  ENGINEERING,B Tech Fresher,AutoCAD,Degree,PLC,...   
73766      0 Reviews  Lecturer Activities,Nursing Education,Trainer,...   

       min experience  max experience     min_salary     max_salary  
0                   1               6  Not disclosed  Not disclosed  
1                  11              15  Not disclosed  Not disclosed  
2                   3               7  Not disclosed  Not disclosed  
3                   3               7  Not disclosed  Not disclosed  
4                   5               7  Not disclosed  Not disclosed  
...               ...             ...            ...            ...  
73762               1               5         700000         850000  
73763               1               5         700000         850000  
73764               1               4         300000         450000  
73765               0               2  Not disclosed  Not disclosed  
73766               1               6  Not disclosed  Not disclosed  

[72988 rows x 11 columns]

In [ ]:
df.info()

In [ ]:
df['min_salary'].unique()

### **2.6 Clean Location Column**

In [ ]:
df['location']

In [ ]:
for i in range(50):
  print(df['location'][i])

In [ ]:
df['location'] = df['location'].str.replace('(All Areas)', '')
df['location']

In [ ]:
for i in range(50):
  print(df['location'][i])

In [ ]:
df['location'] = df['location'].str.replace(',','-')

In [ ]:
df['location'][0]

In [ ]:
df['location'] = df['location'].str.replace('/','-')

In [ ]:
df['location']

In [ ]:
for i in range(10):
  print(df['location'][i])

In [ ]:
df['clean_location'] = df['location'].str.replace(' ','').str.title().str.split('-')

In [ ]:
df['clean_location']

In [ ]:
df

In [ ]:
del df['location']

In [ ]:
df

### **2.7 Clean reviews column**

In [ ]:
df['reviews']

In [ ]:
df['reviews'].unique()

In [ ]:
df['reviews'].nunique()

In [ ]:
df['reviews'].value_counts()

## **3. Data Analysis**

### **3.1 No. of jobs as per experience**